librz/core/cmd_debug.c

```diff
@@ -1443,376 +1370,465 @@ static bool get_bin_info(RzCore *core, const char *file, ut64 baseaddr, PJ *pj,
 	return true;
 }
 
-static int cmd_debug_map(RzCore *core, const char *input) {
+// dm
+RZ_IPI RzCmdStatus rz_cmd_debug_list_maps_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	if (rz_debug_is_dead(core->dbg)) {
+		rz_cons_println("Debugging is not enabled. Run ood?");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_debug_map_sync(core->dbg); // update process memory maps
+	rz_debug_map_print(core->dbg, core->offset, mode);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dma
+RZ_IPI RzCmdStatus rz_cmd_debug_allocate_maps_handler(RzCore *core, int argc, const char **argv) {
+	if (rz_debug_is_dead(core->dbg)) {
+		rz_cons_println("Debugging is not enabled. Run ood?");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	ut64 addr = core->offset;
+	int size = (int)rz_num_math(core->num, argv[1]);
+	rz_debug_map_alloc(core->dbg, addr, size, false);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dmm
+RZ_IPI RzCmdStatus rz_cmd_debug_modules_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	if (rz_debug_is_dead(core->dbg)) {
+		rz_cons_println("Debugging is not enabled. Run ood?");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (!strcmp(argv[0] + 3, ".*")) {
+		cmd_debug_modules(core, ':');
```

Comment:
Could you also please tear `cmd_debug_modules` function apart and use the API instead of passing characters/string further?

Reply:
I can see that is uses `int mode` and matching characters instead. I suggest to extract `?` to drop the help, it should be provided by the newshell anyway, and use `RzOutputMode mode` directly.